src/components/motorcycleForm/MotorcycleForm.jsx

```diff
@@ -0,0 +1,126 @@
+import { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import './motorcycleForm.css';
+import {
+  FloatingLabel, Form, InputGroup, Button, Row, Col,
+} from 'react-bootstrap';
+import { useNavigate } from 'react-router-dom';
+import { createMotorcycle } from '../../redux/actions/motorcycleActions';
+
+function MotorcycleForm() {
+  const [model, setModel] = useState(null);
+  const [description, setDescription] = useState(null);
+  const [photo, setPhoto] = useState(null);
+  const [duration, setDuration] = useState(1);
+  const [financeFee, setFinanceFee] = useState(0);
+  const [purchaseFee, setPurchaseFee] = useState(0);
+  const [amountPayable, setAmountPayable] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+  const error1 = useSelector((state) => state.motorcycles.error);
+  const error2 = useSelector((state) => state.users.loading);
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+    const formData = new FormData();
+    formData.append('motorcycle[model]', model);
+    formData.append('motorcycle[description]', description);
+    formData.append('motorcycle[duration]', duration);
+    formData.append('motorcycle[finance_fee]', financeFee);
+    formData.append('motorcycle[purchase_fee]', purchaseFee);
+    formData.append('motorcycle[amount_payable]', amountPayable);
+    formData.append('motorcycle[photo]', photo);
+    const isSuccess = await dispatch(createMotorcycle(formData));
+    if (isSuccess.meta.requestStatus === 'fulfilled') { navigate('/motorcycles'); }
```

Comment:
I think we can go to the reservations route after this post or I can add it in my next in coming task

Reply:
Redirecting to the reservations is better. If you can add it to your next task, that would be great.